src/com/facebook/buck/step/fs/FileScrubberStep.java

```diff
@@ -64,7 +64,6 @@ public StepExecutionResult execute(ExecutionContext context)
       }
     } catch (FileContentsScrubber.ScrubException e) {
       context.logError(e, "Error scrubbing non-deterministic metadata from %s", filePath);
-      return StepExecutionResults.ERROR;
```

Comment:
deterministic builds are extremely important for most of Buck users and one of the core principles behind Buck's philosophy, so this change makes me very uncomfortable. If cgo binaries are important for your project, please consider fixing the scrubber, if not, consider getting rid of them. As a compromise, we can consider adding a flag that controls the behavior in case of scrubber failures which preserves the existing behavior of failing by default but can be switched to a forgiving mode.

Reply:
It looks like the srubber fails on cgo binaries because it doesn't find the absolute paths for the dynamic linker that are present in other Mach-O binaries. I think I can fix that instead. Currently the scrubber fails hard if it doesn't find anything to scrub, in this instance would it be acceptable to succeed if it doesn't find what it's looking for?

Reply:
@monty-uber, I think in such case it would be best to have a separate scrubber for cgo binaries which does not have to fail in case it does not find anything to scrub :)